sql/src/main/java/io/crate/metadata/rule/ingest/IngestRulesMetaData.java

```diff
@@ -64,6 +80,34 @@ public IngestRulesMetaData(StreamInput in) throws IOException {
         }
     }
 
+    @Nullable
+    public Set<IngestRule> getIngestRules(String sourceIdent) {
+        return sourceIngestRules.get(sourceIdent);
+    }
+
+    public void createIngestRule(String sourceIdent, IngestRule ingestRule) throws IllegalArgumentException {
+        for (Set<IngestRule> ingestRules : sourceIngestRules.values()) {
+            if (ingestRules.contains(ingestRule)) {
+                throw new IllegalArgumentException("Ingest rule with name " + ingestRule.getName() + " already exist");
+            }
+        }
+
+        Set<IngestRule> ingestRules = sourceIngestRules.computeIfAbsent(sourceIdent, k -> new HashSet<>());
+        ingestRules.add(ingestRule);
+    }
+
+    public long dropIngestRule(String ruleName) {
+        for (Set<IngestRule> ingestRules : sourceIngestRules.values()) {
+            for (IngestRule rule : ingestRules) {
+                if (rule.getName().equals(ruleName)) {
+                    ingestRules.remove(rule);
+                    return 1L;
+                }
+            }
+        }
+        return 0L;
```

Comment:
shouldn't we raise an exception here if the rule is not found? then no return value is needed at all